plugins/inputs/burrow/burrow.go

```diff
@@ -0,0 +1,349 @@
+	URI     string `json:"url"`
+	Host    string `json:"host"`
+	Cluster string `json:"cluster"`
+	Group   string `json:"group"`
+	Topic   string `json:"topic"`
+}
+type HTTPResponseError struct {
+	Error   bool                      `json:"error"`
+	Message string                    `json:"message"`
+	Result  map[string]string         `json:"result"`
+	Request BurrowResponseRequestInfo `json:"request"`
+}
+
+type BurrowResponseClusterList struct {
+	Error    bool                      `json:"error"`
+	Message  string                    `json:"message"`
+	Clusters []string                  `json:"clusters"`
+	Request  BurrowResponseRequestInfo `json:"request"`
+}
+
+type BurrowResponseTopicList struct {
+	Error   bool                      `json:"error"`
+	Message string                    `json:"message"`
+	Topics  []string                  `json:"topics"`
+	Request BurrowResponseRequestInfo `json:"request"`
+}
+
+type BurrowResponseTopicDetail struct {
+	Error   bool                      `json:"error"`
+	Message string                    `json:"message"`
+	Offsets []int64                   `json:"offsets"`
+	Request BurrowResponseRequestInfo `json:"request"`
+}
+
+type BurrowResponseConsumerList struct {
+	Error     bool                      `json:"error"`
+	Message   string                    `json:"message"`
+	Consumers []string                  `json:"consumers"`
+	Request   BurrowResponseRequestInfo `json:"request"`
+}
+
+type BurrowResponseConsumerStatus struct {
+	Error   bool                      `json:"error"`
+	Message string                    `json:"message"`
+	Status  ConsumerGroupStatus       `json:"status"`
+	Request BurrowResponseRequestInfo `json:"request"`
+}
+
+type Burrow struct {
+	client *http.Client
+
+	Urls     []string
+	Clusters []string
+	Topics   []string
+	Groups   []string
+}
+
+var sampleConfig = `
+  ## Burrow HTTP endpoint urls.
+  urls = ["http://burrow-service.com:8000"]
+  ## Clusters to fetch data. Default to fetch all.
+  #clusters = []
+  ## Topics to monitor. Default to monitor all from Burrow.
+  #topics = []
+  ## Groups to monitor. Default to monitor all from Burrow.
+  #groups = []
+`
+
+func (b *Burrow) SampleConfig() string {
+	return sampleConfig
+}
+
+func (b *Burrow) Description() string {
+	return "Collect Kafka topics and consumers status from Burrow's (https://github.com/linkedin/Burrow) HTTP Endpoint."
+}
+
+func (b *Burrow) Gather(acc telegraf.Accumulator) error {
+	if b.client == nil {
+		b.client = &http.Client{
+			Transport: tr,
+			Timeout:   time.Duration(3 * time.Second),
+		}
+	}
+
+	var wg sync.WaitGroup
+	errChan := errchan.New(len(b.Urls))
+
+	for _, u := range b.Urls {
+		addr, err := url.Parse(u)
+		if err != nil {
+			return fmt.Errorf("Unable to parse address '%s': %s", u, err)
+		}
+
+		wg.Add(1)
+		go func(addr *url.URL) {
+			defer wg.Done()
+			errChan.C <- b.gatherUrl(addr, acc)
+		}(addr)
+	}
+
+	wg.Wait()
+	return errChan.Error()
+}
+
+var tr = &http.Transport{
```

Comment:
Create the transport at the same time you instantiate the client, this allows differing settings across multiple plugin instances.